executor/adapter.go

```diff
@@ -323,6 +324,19 @@ func (a *ExecStmt) setPlanLabelForTopSQL(ctx context.Context) context.Context {
 	return topsql.AttachSQLInfo(ctx, normalizedSQL, sqlDigest, normalizedPlan, planDigest, vars.InRestrictedSQL)
 }
 
+func (a *ExecStmt) countExecForTopSQL() {
+	if !variable.TopSQLEnabled() {
+		return
+	}
+	vars := a.Ctx.GetSessionVars()
+	_, sqlDigest := vars.StmtCtx.SQLDigest()
+	_, planDigest := vars.StmtCtx.GetPlanDigest()
+	if vars.StmtStats != nil {
+		vars.StmtStats.AddExecCount(sqlDigest.String(), planDigest.String(), time.Now().Unix(), 1)
```

Comment:
> Is it a right place to put StmtStats in the "var"? It is never a variable I think.

The idea is:
1. Initially it is located in the `session struct`, but we found that we cannot count execution in the session module because we need the parsed plan digest.
2. Then we copied a StmtStats pointer in SessionVars, the purpose is to be able to access it in the executor.
3. In the end we found that StmtStats in `session struct` is no longer necessary, we can maintain only one pointer in `SessionVars`.

Reply:
> Is it possible to not pass in `time.Now().Unix()`? Is there any case that you need to customize it to be not `time.Now().Unix()`?

One possible scenario is: we must count at this position, but want to record an earlier timestamp (e.g. request just arrived, no parse yet).

Reply:
I would 

> > Is it possible to not pass in `time.Now().Unix()`? Is there any case that you need to customize it to be not `time.Now().Unix()`?
> 
> One possible scenario is: we must count at this position, but want to record an earlier timestamp (e.g. request just arrived, no parse yet).

I would like to forbid this behavior. It means even when a series of data is already written, there will be future new writes that changed the order of this series, inserting a new data in the middle of them. The data is no longer append-only pattern for a single series.

Reply:
I think it's ok since `StatementContext` also in `SessionVars`

Reply:
> > > Is it possible to not pass in `time.Now().Unix()`? Is there any case that you need to customize it to be not `time.Now().Unix()`?
> > 
> > 
> > One possible scenario is: we must count at this position, but want to record an earlier timestamp (e.g. request just arrived, no parse yet).
> 
> I would like to forbid this behavior. It means even when a series of data is already written, there will be future new writes that changed the order of this series, inserting a new data in the middle of them. The data is no longer append-only pattern for a single series.

We are discussing removing the mode where the timestamp is specified by the user.